src/app/dashboard/dashboard.component.ts

```diff
@@ -113,35 +129,43 @@ export class DashboardComponent implements OnInit {
       dataSource: this.dataSource,
     };
 
-    this.userService.addUser(this.user).subscribe((user) => {
-      this.users.push(user);
-    });
+    console.log(this.user);
+
+    this.subscriptions.add(
+      this.userService.addUser(this.user).subscribe((user) => {
+        this.users.push(user);
+      })
+    );
 
     this.resetForm();
   }
 
   deleteUser(user: User) {
-    this.users = this.users.filter((u) => u !== user);
-    this.userService.deleteUser(user.id).subscribe(() => {
-      this.messageService.add({
-        severity: 'success',
-        summary: 'Success',
-        detail: 'User Deleted!',
-      });
-    });
-  }
-
-  updateUser() {
-    if (this.user) {
-      this.userService.updateUser(this.user).subscribe(() => {
-        const index = this.users.findIndex((u) => u.id === this.user.id);
-        this.users[index] = this.user;
+    _.remove(this.users, (u) => u === user);
+    this.subscriptions.add(
+      this.userService.deleteUser(user.id).subscribe(() => {
         this.messageService.add({
           severity: 'success',
           summary: 'Success',
-          detail: 'User Updated!',
+          detail: 'User Deleted!',
         });
-      });
+      })
+    );
+  }
+
+  updateUser() {
+    if (this.user) {
+      this.subscriptions.add(
+        this.userService.updateUser(this.user).subscribe(() => {
+          this.users[this.users.findIndex((u) => u.id === this.user.id)] =
+            this.user;
+          this.messageService.add({
+            severity: 'success',
+            summary: 'Success',
+            detail: 'User Updated!',
+          });
```

Comment:
indentations

Reply:
pano pong indentations? should I use 4 spaces na indents?? Kung ganon po I'll enforce eslint + prettier nlng para proper code structure po yung ma format.